README.md

```diff
@@ -1,196 +1,289 @@
+For a workflow the `executable_input` hash will be prepoluated with the key-value pair {"`0.upload_sentinel_record`": `$record_id`} where `$record_id` is the DNAnexus file-id of the sentinel record generated for the uploaded RUN directory (see section titled **Files generated**).
+**It is the user's responsibility to ensure that the specified applet/workflow has an appropriate input contract which accepts a DNAnexus record with the input name of `upload_sentinel_record`**
+Additional input/options can be specified, statically using the Ansible variable `downstream_input`. This should be provided as a JSON string, parsable, at the top level, as a Python dict of `str` to `str`. 
+Example of a properly formatted `downstream_input` for an `applet`
+- ```{"input_name1": "value1", "input_name2": "value2"}```
+Example of a properly formatted `downstream_input` for a `workflow`
+- ```{"0.step0_input": "value1", "1.step2_input": "value2"})```
+*Note the numerical index prefix necessary when specifying input for an `workflow`, which disambiguates which step in the workflow an input is targeted to*
+#### Files generated
+We use a hypothetical example of a local RUN folder named `20160101_M000001_0001_000000000-ABCDE`, that was placed into the `monitored_directory`, after the `dx-streaming-upload` role has been set up.
+**Local Files Generated**
+```
+path/to/LOG/directory
+(specified in monitor_run_config.template file)
+- 20160101_M000001_0001_000000000-ABCDE.lane.all.log
+path/to/TMP/directory
+(specified in monitor_run_config.template file)
+- no persistent files (tar files stored transiently, deleted upon successful upload to DNAnexus)
+```
+**Files Streamed to DNAnexus project**
+```
+project
+   └───20160101_M000001_0001_000000000-ABCDE
+       │───runs
+       │    │  RunInfo.xml
+       │    │  SampleSheet.csv
+       │    │  run.20160101_M000001_0001_000000000-ABCDE.lane.all.log
+       │    │  run.20160101_M000001_0001_000000000-ABCDE.lane.all.upload_sentinel
+       │    │  run.20160101_M000001_0001_000000000-ABCDE.lane.all_000.tar.gz
+       │    │  run.20160101_M000001_0001_000000000-ABCDE.lane.all_001.tar.gz
+       │    │  ...
+       │
+       └───reads (or analyses)
+            │  output files from downstream applet (e.g. demx)
+            │   "reads" folder will be created if an applet is triggered
+            │   "analyses" folder will be created if a workflow is triggered
+            │  ...
+```
+The `reads` folder (and subfolders) will only be created if `applet` is specified.
+The `analyses` folder (and subfolder) will only be created if `workflow` is specified.
+`RunInfo.xml` and `SampleSheet.csv` will only be upladed if they can be located within the root of the local RUN directory.

```

Comment:
--> " `RunInfo.xml` and `SampleSheet.csv` will only be uploaded if they can be located within the root of the local RUN directory. "